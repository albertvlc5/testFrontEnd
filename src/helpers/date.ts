export const getPreviousMonthNameByLocale = (
  dateStr: string,
  locale = 'pt-BR',
) => {
  const date = new Date(dateStr);
  const month = date.getUTCMonth();
  const previousMonth = month === 0 ? 11 : month - 1;

  const monthName = new Date(
    date.getUTCFullYear(),
    previousMonth,
    1,
  ).toLocaleString(locale, { month: 'long' });

  return monthName;
};
