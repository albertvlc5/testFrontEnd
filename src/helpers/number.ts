import 'intl';
import 'intl/locale-data/jsonp/pt';
export const formatNumberByLocale = (value: number, locale = 'pt-Br'): string =>
  new Intl.NumberFormat(locale, { minimumFractionDigits: 2 }).format(value);

export const formatMoney2Numeral = (money: string): number => {
  const [value, cents] = String(money).split(',');
  return Number([value.replace(/\D+/g, ''), cents].join('.'));
};

export const numberToCurrencyIntl = (
  value: number = 0,
  locale = 'pt-BR',
  currency = 'BRL',
) =>
  new Intl.NumberFormat(locale, { style: 'currency', currency }).format(value);
