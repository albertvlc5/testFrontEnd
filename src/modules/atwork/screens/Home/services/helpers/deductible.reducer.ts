import {
  IDeductible,
  IDeductibleTransformed,
} from '../../interfaces/deductible.interface';

export const initialDeductibleTransformed = {
  totalDiscounts: 0,
  dueDate: '',
};

export const deductibleReducer = (
  acumulator: IDeductibleTransformed,
  item: IDeductible,
) => ({
  totalDiscounts: acumulator.totalDiscounts + item.amountDeductibleThisMonth,
  dueDate: item.dueDate,
});
