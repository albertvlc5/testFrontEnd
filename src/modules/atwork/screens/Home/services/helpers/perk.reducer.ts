import { ProductType } from '../../interfaces/product-type.enum';
import { IPerk, IPerkTransformed } from '../../interfaces/perk.interface';

export const initialPerkTransformed = {
  totalAmount: 0,
  salaryAdvancement: {
    totalAmountAvailable: 0,
    amountDeductible: 0,
    currency: '',
  },
  payroll: {
    amountDeductible: undefined,
    totalAmountAvailable: 0,
    currency: '',
  },
};

export const perkReducer = (acumulator: IPerkTransformed, item: IPerk) => {
  const salaryAdvancement =
    item.type === ProductType.SALARY_ADVANCEMENT
      ? {
          totalAmountAvailable: item.totalAmountAvailable,
          amountDeductible: item.amountDeductible,
          currency: item.symbol,
        }
      : acumulator.salaryAdvancement;

  const payroll =
    item.type === ProductType.PAYROLL
      ? {
          amountDeductible: item.amountDeductible,
          currency: item.symbol,
          totalAmountAvailable: item.totalAmountAvailable,
        }
      : acumulator.payroll;

  return {
    totalAmount: acumulator.totalAmount + item.totalAmountAvailable,
    salaryAdvancement: salaryAdvancement,
    payroll: payroll,
  };
};
