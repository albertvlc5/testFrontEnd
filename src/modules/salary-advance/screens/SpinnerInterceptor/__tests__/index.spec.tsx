import React from 'react';
import { mockedNavigation } from '../../../../../../src/__tests__/mocks';
import { mountWithAtworkHost } from '../../../../../../src/__tests__/helpers';
import { SpinnerInterceptor } from '../index';

jest.mock('../../../../../CoreService', () => ({
  __esModule: true,
  default: {
    getInstanceV1: () => ({
      session: {
        getUserSession: jest.fn().mockReturnValue(
          Promise.resolve({
            profile: {},
            token: {},
            auth: {},
            apiVersion: '1',
          }),
        ),
      },
    }),
  },
}));

describe('Spinner Intercept Screen', () => {
  it('should render component', () => {
    const wrapper = mountWithAtworkHost(
      <SpinnerInterceptor navigation={mockedNavigation} />,
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
