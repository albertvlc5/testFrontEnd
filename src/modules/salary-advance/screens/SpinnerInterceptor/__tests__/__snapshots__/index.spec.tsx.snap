// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Intercept Screen should render component 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    animating={true}
    animationDuration={4000}
    animationEasing={[Function]}
    color="#11bb77"
    hideAnimationDuration={200}
    interaction={true}
    renderComponent={[Function]}
    style={
      Object {
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "height": 50,
            "transform": Array [
              Object {
                "rotate": "82.5deg",
              },
              Object {
                "rotate": "0deg",
              },
            ],
            "width": 50,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "height": 25,
              "overflow": "hidden",
              "width": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 50,
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "rotate": "-172.5deg",
                  },
                ],
                "width": 50,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "height": 25,
                  "overflow": "hidden",
                  "width": 50,
                }
              }
            >
              <View
                style={
                  Object {
                    "borderColor": "#11bb77",
                    "borderRadius": 25,
                    "borderWidth": 5,
                    "height": 50,
                    "width": 50,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "height": 50,
            "transform": Array [
              Object {
                "rotate": "82.5deg",
              },
              Object {
                "rotate": "0deg",
              },
            ],
            "width": 50,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "height": 25,
              "overflow": "hidden",
              "top": 25,
              "width": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 50,
                "transform": Array [
                  Object {
                    "translateY": -25,
                  },
                  Object {
                    "rotate": "352.5deg",
                  },
                ],
                "width": 50,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "height": 25,
                  "overflow": "hidden",
                  "width": 50,
                }
              }
            >
              <View
                style={
                  Object {
                    "borderColor": "#11bb77",
                    "borderRadius": 25,
                    "borderWidth": 5,
                    "height": 50,
                    "width": 50,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
